src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -419,6 +419,14 @@ protected function getEnv($name)
         if (false !== $env = getenv($name)) {
             return $this->envCache[$name] = $env;
         }
+        if (false !== $i = strpos($name, '@')) {
+            $id = substr($name, 1 + $i);
+            $service = isset($this->services[$id]) ? $this->services[$id] : (isset($this->methodMap[$id]) ? $this->{$this->methodMap[$id]}() : $this->get($id));
+
+            if (null !== $env = $service->getEnv(substr($name, 0, $i))) {
+                return $this->envCache[$name] = $env;
+            }
+        }
         if (!$this->hasParameter("env($name)")) {
```

Comment:
Shouldnt we use the normalized name here? And provide a default regarding the lookup service?